TileStache/Mapnik.py

```diff
@@ -74,22 +74,37 @@ class ImageProvider:
         - http://trac.mapnik.org/wiki/XMLConfigReference
     """
 
-    def __init__(self, layer, mapfile, fonts=None, scale_factor=None):
-        """ Initialize Mapnik provider with layer and mapfile.
+    def __init__(self, layer, mapconfig, fonts=None, scale_factor=None):
+        """ Initialize Mapnik provider with layer and mapconfig.
 
-            XML mapfile keyword arg comes from TileStache config,
-            and is an absolute path by the time it gets here.
+            XML mapconfig keyword arg comes from TileStache config,
+            and is either an absolute path by the time it gets here or a
+            XML string.
         """
-        maphref = urljoin(layer.config.dirpath, mapfile)
-        scheme, h, path, q, p, f = urlparse(maphref)
+        self.mapnik = None
+        self.layer = layer
 
-        if scheme in ('file', ''):
-            self.mapfile = path
-        else:
-            self.mapfile = maphref
+        try:
```

Comment:
No, because later we will break backwards compatibility. We need to do it the right way from the beginning. Sure, devs don't need to look this code, but if something fails, we can't tell the user if the XML was not well formatted or the path was wrong, because we don't have a way to properly differentiate between the two possible cases.